bc_obps/compliance/service/automated_process/compliance_handlers.py

```diff
@@ -0,0 +1,110 @@
+        """Handle the compliance update."""
+        ...
+
+
+class PenaltyPaidHandler(ComplianceUpdateHandler):
+    """Handling invoices with penalties that are fully paid."""
+
+    def can_handle(self, invoice: ElicensingInvoice, obligation: ComplianceObligation) -> bool:
+        """Check if invoice has a penalty and the penalty invoice is fully paid."""
+        has_penalty = hasattr(invoice, 'compliance_penalty') and getattr(invoice, 'compliance_penalty', None)
+        if not has_penalty:
+            return False
+
+        penalty_record = getattr(invoice, 'compliance_penalty', None)
+        if not penalty_record or not penalty_record.elicensing_invoice:
+            return False
+
+        penalty_invoice = penalty_record.elicensing_invoice
+        return penalty_invoice and penalty_invoice.outstanding_balance == Decimal('0.00')
+
+    def handle(self, invoice: ElicensingInvoice, obligation: ComplianceObligation) -> None:
+        """Update obligation penalty_status to PAID if penalty invoice is fully paid."""
+        if obligation.penalty_status != ComplianceObligation.PenaltyStatus.PAID:
+            ComplianceObligationService.update_penalty_status(obligation.pk, ComplianceObligation.PenaltyStatus.PAID)
+            logger.info(f"Updated penalty status to PAID for obligation {obligation.obligation_id}")
+
+
+class PenaltyAccruingHandler(ComplianceUpdateHandler):
+    """Handler for handling obligations that need to start accruing penalties."""
+
+    def can_handle(self, invoice: ElicensingInvoice, obligation: ComplianceObligation) -> bool:
+        """Check if obligation should start accruing penalties."""
+        has_penalty = hasattr(invoice, 'compliance_penalty') and getattr(invoice, 'compliance_penalty', None)
+        if has_penalty:
+            return False
+
+        return (
+            obligation.compliance_report_version.status == ComplianceReportVersion.ComplianceStatus.OBLIGATION_NOT_MET
+            and invoice.outstanding_balance > Decimal('0.00')
+            and invoice.due_date < timezone.now()
+        )
+
+    def handle(self, invoice: ElicensingInvoice, obligation: ComplianceObligation) -> None:
+        """Update penalty status to ACCRUING and create penalty if needed."""
+        if obligation.penalty_status != ComplianceObligation.PenaltyStatus.ACCRUING:
+            ComplianceObligationService.update_penalty_status(
+                obligation.pk, ComplianceObligation.PenaltyStatus.ACCRUING
+            )
+            logger.info(f"Updated penalty status to ACCRUING for obligation {obligation.obligation_id}")
+
+        if not getattr(obligation, 'compliance_penalty', None):
+            PenaltyCalculationService.create_penalty(obligation)
+            logger.info(f"Created penalty for obligation {obligation.obligation_id}")
+        else:
+            logger.info(f"Penalty already exists for obligation {obligation.obligation_id}")
```

Comment:
We can remove this section. Updating the status to "ACCRUING" in this handler is all we need to do since we do not create penalty records/invoices until an obligation is fully paid.